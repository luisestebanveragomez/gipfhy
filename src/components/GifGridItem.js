import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {

  return (
      <div className="card">
          <img src={ url } alt={ title } />
          <strong> { title } </strong>
      </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
