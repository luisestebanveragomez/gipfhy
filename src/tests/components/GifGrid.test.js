import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('pruebas en <GifGrid />', () => {

  const text = 'tomalo';

  test('debe de mostrarse correctamente', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })

    const wrapper = shallow(<GifGrid category={ text }  />);
    expect(wrapper).toMatchSnapshot();
  })

  test('debe de mostrar items cuando se cargara imÃ¡genes useFetchGifs', () => {
    const gifs = [{
      id: '123',
      url: 'hey',
      title: 'texto'
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })

    const wrapper = shallow(<GifGrid category={ text }  />);
    expect( wrapper.find('p').exists() ).toBeFalsy();
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
  })
})