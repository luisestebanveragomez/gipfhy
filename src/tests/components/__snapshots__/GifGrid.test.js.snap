// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3>
     
    tomalo
     
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
